rageval/metrics/_answer_chrf.py

```diff
@@ -0,0 +1,179 @@
+    author = "Popovi{\'c}, Maja",
+    booktitle = "Proceedings of the Second Conference on Machine Translation",
+    month = sep,
+    year = "2017",
+    address = "Copenhagen, Denmark",
+    publisher = "Association for Computational Linguistics",
+    url = "https://aclanthology.org/W17-4770",
+    doi = "10.18653/v1/W17-4770",
+    pages = "612--618",
+}
+@inproceedings{post-2018-call,
+    title = "A Call for Clarity in Reporting {BLEU} Scores",
+    author = "Post, Matt",
+    booktitle = "Proceedings of the Third Conference on Machine Translation: Research Papers",
+    month = oct,
+    year = "2018",
+    address = "Belgium, Brussels",
+    publisher = "Association for Computational Linguistics",
+    url = "https://www.aclweb.org/anthology/W18-6319",
+    pages = "186--191",
+}
+"""
+
+
+@dataclass
+@add_attribute('mtype', 'AnswerCorrectness')
+@datasets.utils.file_utils.add_start_docstrings(_DESCRIPTION, _KWARGS_DESCRIPTION)
+class AnswerCHRFCorrectness(Metric):
+    """Estimates the CHRF between answers and ground truth answers."""
+
+    name = "answer_chrf"
+
+    ALIAS = ['answer_chrf']
+
+    def __init__(
+            self,
+            char_order: int = 6,
+            word_order: int = 0,
+            beta: int = 2,
+            lowercase: bool = False,
+            whitespace: bool = False,
+            eps_smoothing: bool = False
+    ):
+        """
+        Explicitly initialize AnswerCHRFCorrectness.
+
+        Ensure all parent classes are initialized.
+        """
+        super().__init__()
+        self._required_columns = ['answers', 'gt_answers']
+        self.char_order = char_order
+        self.word_order = word_order
+        self.beta = beta
+        self.lowercase = lowercase
+        self.whitespace = whitespace
+        self.eps_smoothing = eps_smoothing
+
+    def __repr__(self) -> str:
+        """:return: Formatted string representation of the metric."""
+        return f"{self.ALIAS[0]}"
+
+    def _info(self):
+        return datasets.MetricInfo(
+            description=_DESCRIPTION,
+            inputs_description=_KWARGS_DESCRIPTION,
+            citation=_CITATION,
+            features=datasets.Features(
+                {
+                    "answers": datasets.Value("string"),
+                    "gt_answers": datasets.Value("string")
+                }
+            )
+        )
+
+    def _validate_data(self, dataset: datasets.Dataset) -> bool:
+        """Validate the of the input dataset."""
+        super()._validate_data(dataset)
+        if not all(isinstance(answer, str) for answer in dataset["answers"]):
+            raise ValueError("The type of answers should be a string.")
+        if not all(isinstance(a, List) or not all(isinstance(item, str) for item in a) for a in dataset["gt_answers"]):
+            raise ValueError("The type of gt_answers should be a list of strings.")
+
+    def compute(
+        self,
+        dataset: Dataset,
+        batch_size: int = None,
```

Comment:
In ter or chrf metric, the final score is not the average of the score of each sample, so we need to process the whole dataset to get the final score instead of spliting the dataset into batches.

Reply:
OK